### DISCUSSION TITLE
Runtime error: memory access out of bounds

### DISCUSSION THREAD

**Discussion by Andrea332** [jeffreylanters/react-unity-webgl#425] [Help Wanted!]
Simple Build with one scene i get this error on mobile devices and is extremely laggy : Runtime error: memory access out of bounds

**Comment by jeffreylanters** [comment]
It sounds like your canvas is getting too large. Some Unity versions contain a bug where the Unity canvas grows indefinitely when no sizing constraints are provided. Have you tried providing a width and height to the Unity component?

**Comment by Andrea332** [comment]
Current Canvas that i'm using

![image](https://user-images.githubusercontent.com/74671832/183632279-bd82838c-e114-40fb-929e-9af701894a16.png)

**Reply by jeffreylanters** [reply]
Sorry, I meant the HTML canvas. Try providing a height and width to it. 

`<Unity style={{ height: 600, width: 800 }} .../>`

**Comment by Andrea332** [comment]
Could be something related to this coroutine that i start at the beginning of the scene?:

public IEnumerator LoadSceneAsynchronously(string sceneName)
        {
            sceneLoadingStart?.Invoke();
            asyncOperation = SceneManager.LoadSceneAsync(sceneName);
            while (!asyncOperation.isDone)
            {
                sceneLoadingProgress?.Invoke(asyncOperation.progress);
                Debug.Log(asyncOperation.progress);
                yield return null;
            }
        }
